pkg/java/v1/scaffolds/api.go

```diff
@@ -2,18 +2,31 @@ package scaffolds
 
 import (
 	"fmt"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins"
 
 	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
 
-	"github.com/java-operator-sdk/kubebuilder-plugin/pkg/internal/kubebuilder/cmdutil"
 )
 
 type apiScaffolder struct {
 	fs machinery.Filesystem
+
+	config   config.Config
+	resource resource.Resource
+
+	doRole, doPlaybook bool
 }
 
-func NewCreateAPIScaffolder() cmdutil.Scaffolder {
-	return &apiScaffolder{}
+// NewCreateAPIScaffolder returns a new plugins.Scaffolder for project initialization operations
+func NewCreateAPIScaffolder(cfg config.Config, res resource.Resource, doRole, doPlaybook bool) plugins.Scaffolder {
+	return &apiScaffolder{
+		config:     cfg,
+		resource:   res,
+		doRole:     doRole,
+		doPlaybook: doPlaybook,
+	}
```

Comment:
role and playbook are ansible specific, remove them.
```suggestion
func NewCreateAPIScaffolder(cfg config.Config, res resource.Resource) plugins.Scaffolder {
	return &apiScaffolder{
		config:     cfg,
		resource:   res,
	}
```

Reply:
Modified according to suggestion.